### PR TITLE
extended fprotect - RRAMC combined regions

### PR DESCRIPTION
**Pull request by michalek-no**
Adds option to lock across both regions.

### CODE REVIEW

**Comment by michalek-no** [comment]
rebase

File: sysbuild/Kconfig.mcuboot [hunk 1]
```diff
@@ -120,6 +120,11 @@ endchoice
 
 endif
 
+config FPROTECT_ALLOW_COMBINED_REGIONS
+	bool
+	default y
+	depends on (BOOTLOADER_MCUBOOT && !SECURE_BOOT_APPCORE)
```
Comment (nordicjm) [inline sysbuild/Kconfig.mcuboot:126]:
```suggestion
	depends on SOC_SERIES_NRF54LX && !SECURE_BOOT_APPCORE
```
move the above `endif` to below this so it's included in the if check

File: sysbuild/CMakeLists.txt [hunk 2]
```diff
@@ -140,6 +140,9 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
 
 # TODO: SB_CONFIG_SOC_SERIES_NRF54HX
 
+  if(SB_CONFIG_FPROTECT_ALLOW_COMBINED_REGIONS)
+    set_config_bool(mcuboot CONFIG_FPROTECT_ALLOW_COMBINED_REGIONS y)
+  endif()
```
Comment (nordicjm) [inline sysbuild/CMakeLists.txt:145]:
```suggestion
  if(SB_CONFIG_MCUBOOT_FPROTECT_ALLOW_COMBINED_REGIONS)
    set_config_bool(mcuboot CONFIG_FPROTECT_ALLOW_COMBINED_REGIONS y)
  else()
    set_config_bool(mcuboot CONFIG_FPROTECT_ALLOW_COMBINED_REGIONS n)
  endif()
```
Move to line 336, just above `# Apply image compression support options`

File: lib/fprotect/Kconfig [hunk 3]
```diff
@@ -72,4 +72,11 @@ config FPROTECT_APP
 	  'pm_config.h' in your build directory, or by invoking the
 	  'partition_manager_report' target.
 
+config FPROTECT_ALLOW_COMBINED_REGIONS
+	bool "Use both RRAMC_regions"
+	depends on SOC_SERIES_NRF54LX
```
Comment (nordicjm) [inline lib/fprotect/Kconfig:77]:
```suggestion
	depends on SOC_SERIES_NRF54LX && MCUBOOT
```

**Review by nordicjm — COMMENTED** [review]
Looks ok

**Review by nvlsianpu — APPROVED** [review]

**Comment by michalek-no** [comment]
rebase

**Comment by kapi-no** [comment]
Please label it "backport v2.8-branch" before the merge if you want to have this change in NCS 2.8

**Comment by michalek-no** [comment]
> Please label it "backport v2.8-branch" before the merge if you want to have this change in NCS 2.8

done, CI green.
When can we merge?

**Review by nordicjm — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**